rakelib/boxname.rake

```diff
@@ -0,0 +1,162 @@
+      # simp-metadata-based ISO builds.
+      #
+      def semver_simpbox_hack_checks(vars_json_data)
+        iso_name = File.basename(vars_json_data['iso_url'])
+        str = iso_name.sub(/^SIMP-/,'').sub(/\.iso$/,'').gsub(/-x86_64/,'.x86-64').gsub(/_/,'-')
+        if (str == semver_fragment(str))
+          str
+        else
+          false
+        end
+      end
+
+      # Make SURE a String is SemVer (but if it's alredy SemVer, leave it alone)
+      #
+      # NOTE: This ought to be a last resort, as it translates things like #
+      #       "6.X" into "6.0.0-X".   If you have any more up your sleeve that
+      #       could infer a more specific version, try them first!
+      #
+      # @param fragment [String] A SemVer fragment to evaluate
+      # @return [String] A valid SemVer String.  If `fragment` was already
+      #         valid SemVer, it will be returned unchanged.
+      #
+      def ensure_semver(fragment)
+        if semver_xyz_count(fragment) < 3
+          str = semver_xyz_match(fragment)
+          suffix = box_simp_release.sub(/^#{str}\.?/,'').sub(/^(.+)$/,'-\1')
+          fragment = "#{pad_missing_semver_xyz_sections(fragment)}#{suffix}"
+        end
+        fragment
+      end
+
+      # Return as much of valid SemVer (including suffixes for pre-relase and
+      #   build metadata as possible) as a String contains
+      #
+      # SemVer 2.0.0 regex, tested at http://rubular.com/r/s9QAkciFhz
+      #
+      # @param str [String] A SemVer fragment to evaluate
+      # @return [String] As much of a valid SemVer String as there is
+      def semver_fragment(str)
+        semver2_0_0ish_regex=/^(((?:\d+\.?){2}\d)(?:[\-+][a-zA-Z0-9\-.]*?$)|((?:\d+\.?){0,2}\d))/
+        semver2_0_0ish_regex.match(str).captures.first
+      end
+
+      # Return as much of a SemVer X.Y.Z number as a String contains
+      #
+      # @param str [String] A SemVer fragment to evaluate
+      # @return [String] Matching SemVer X.Y.Z sections
+      def semver_xyz_match(str)
+        semver2_0_0ish_xyz_regex = /^(((?:\d+\.?){0,2}\d))/
+        matches = semver2_0_0ish_xyz_regex.match(str.to_s)
+        matches.nil? ? '' : matches.captures.first
+      end
+
+      # Count number of SemVer XYZ sections
+      #
+      # @param str [String] A SemVer fragment to evaluate
+      # @return [Integer] Number of SemVer X.Y.Z sections
+      def semver_xyz_count(str)
+        xyz_fragment = semver_xyz_match(str)
+        xyz_fragment.empty? ? 0 : (xyz_fragment.count('.') + 1)
+      end
+
+      # Use zeros to pad any missing SemVer X.Y.Z numbers
+      def pad_missing_semver_xyz_sections(str)
+        count = semver_xyz_count(str)
+        xyz_str = semver_xyz_match(str)
+        count = semver_xyz_count(xyz_str)
+        (3-count).times { xyz_str += '.0' }
+        xyz_str
+      end
+    end
+  end
+end
+
+
+namespace :vagrant do
+  desc <<BOXNAME_DESCRIPTION
+Create boxname.json from SIMP ISO .json file
```

Comment:
What is the boxname.json used for?  Is the Vagrantfile generated by the packer build insufficient?  Also, if this is required, will it be hooked into the box build?

Reply:
@lnemsick-simp This is the Vagrant [box metadata][vbm] used on by `vagrant box add` and beaker to compare box versions and locations.

[vbm]: https://www.vagrantup.com/docs/boxes/format.html#box-metadata